pkg/kgo/source.go

```diff
@@ -1458,11 +1489,18 @@ func (a aborter) trackAbortedPID(producerID int64) {
 // processing records to fetch part //
 //////////////////////////////////////
 
+var rawRecordsPool = pool.NewBucketedPool[kmsg.Record](32, 4096, 2, func(len int) []kmsg.Record {
```

Comment:
why isn't this being injected similar to `decompressor.outBufferPool`?

Reply:
The truth is that I couldn't find any client configuration parameter (something like `max records per batch`) to base the maximum size of `rawRecordsPool` on, so I opted for a generic value that is sufficiently generous and made it globally accessible.